Kernel/VM/PhysicalRegion.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <AK/Retained.h>
+#include <Kernel/Assertions.h>
+#include <Kernel/PhysicalAddress.h>
+#include <Kernel/VM/PhysicalPage.h>
+
+class PhysicalRegion : public Retainable<PhysicalRegion> {
+    AK_MAKE_ETERNAL
+
+    friend class MemoryManager;
+
+public:
+    static Retained<PhysicalRegion> create(PhysicalAddress lower, PhysicalAddress upper);
+    ~PhysicalRegion() {}
+
+    bool is_empty() { return m_next == m_upper; }
```

Comment:
Both is_empty() and size() should be const functions. That makes it possible to call them on a const PhysicalRegion&.

We generally use const for any functions that don’t logically mutate the object.